lib/vm/src/trap/traphandlers.rs

```diff
@@ -708,6 +708,11 @@ impl<'a> CallThreadState<'a> {
             return 1 as *const _;
         }
 
+        // If this fault wasn't in wasm code, then it's not our problem
+        if unsafe { !IS_WASM_PC(pc as _) } {
+            return ptr::null();
+        }
+
```

Comment:
I can confirm that at least on x86_64 mac with the native engine, this is causing the illegal instruction problem

Reply:
I looked into it and reversing this makes all the tests pass 🤔 , that is changing ```if unsafe { !IS_WASM_PC(pc as _) } {```  into ```if unsafe { IS_WASM_PC(pc as _) } {``` makes all the tests pass.

Reply:
I tried changing the default handler to return `true` instead of `false` and then reverting it back to `!IS_WASM_PC(pc as _)` because I suspected that perhaps we weren't actually initializing correctly. This does _not_ solve the problem though. It implies that this is an actual problem with the normal `IS_WASM_PC` logic

Reply:
Yeah, that's because perhaps the Native backend is not pointing to the right function lengths, which is suuuper uber weird

Reply:
Hmm, it's a bit odd that reversing the logic makes it work perfectly 🤔

Reply:
Alright looked into it, the "fix" above causes the JIT engine tests to fail. I added a `dbg` statement in frame_info.rs `module_info` function: the ```        let (end, module_info) = self.ranges.range(pc..).next()?;``` statement always fails with the native engine in tests. So I think you're right, that the ranges for the native engine are all just completely wrong

Reply:
I believe this is now fixed :)

Reply:
Yup, it's now fixed